sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/IsmFormat.java

```diff
@@ -838,59 +743,29 @@ protected long getEncodedElementByteSize(KeyPrefix value, Coder.Context context)
    *   <li>0x01 (version key as a single byte)</li>
    * </ul>
    */
-  static class Footer {
+  @AutoValue
+  abstract static class Footer {
```

Comment:
No changes in visibility as part of the PR. Exposing as public was necessary in BEAM due to package restructuring. See: https://github.com/apache/incubator-beam/commit/e0b11311b287bdb8a303854676bf0f3e37d559c8